kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/InCommitTimestampSuite.scala

```diff
@@ -81,8 +88,126 @@ class InCommitTimestampSuite extends DeltaTableWriteSuiteBase {
 
       val ver1Snapshot = table.getLatestSnapshot(engine).asInstanceOf[SnapshotImpl]
       assertHasWriterFeature(ver1Snapshot, "inCommitTimestamp-preview")
-      assert(
-        getInCommitTimestamp(engine, table, 1).get >= beforeCommitAttemptStartTime)
+      assert(ver1Snapshot.getTimestamp(engine) > ver0Snapshot.getTimestamp(engine))
+      assert(getInCommitTimestamp(engine, table, 1).get == ver1Snapshot.getTimestamp(engine))
+    }
+  }
+
+  test("InCommitTimestamps are monotonic even when the clock is skewed") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val startTime = System.currentTimeMillis()
+      val clock = new ManualClock(startTime)
+      val table = Table.forPath(engine, tablePath)
+
+      appendData(
+        engine,
+        tablePath,
+        isNewTable = true,
+        testSchema,
+        partCols = Seq.empty,
+        data = Seq(Map.empty[String, Literal] -> dataBatches1),
+        clock = clock,
+        tableProperties = Map(IN_COMMIT_TIMESTAMPS_ENABLED.getKey -> "true")
+      )
+
+      val ver1Snapshot = table.getLatestSnapshot(engine).asInstanceOf[SnapshotImpl]
+      val ver1Timestamp = ver1Snapshot.getTimestamp(engine)
+      assert(IN_COMMIT_TIMESTAMPS_ENABLED.fromMetadata(ver1Snapshot.getMetadata))
+
+      clock.setTime(startTime - 10000)
+      appendData(
+        engine,
+        tablePath,
+        data = Seq(Map.empty[String, Literal] -> dataBatches2),
+        clock = clock
+      )
+
+      val ver2Snapshot = table.getLatestSnapshot(engine).asInstanceOf[SnapshotImpl]
+      val ver2Timestamp = ver2Snapshot.getTimestamp(engine)
+      assert(ver2Timestamp == ver1Timestamp + 1)
```

Comment:
We need to verify that the tables are readable. I will merge this, but could you please address it in the next PR?